src/lava/proc/graded/models.py

```diff
@@ -0,0 +1,148 @@
+    """Fixed point implementation of GradedVec"""
+    a_in = LavaPyType(PyInPort.VEC_DENSE, np.int32, precision=24)
+    s_out = LavaPyType(PyOutPort.VEC_DENSE, np.int32, precision=24)
+    vth: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    v: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    exp: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+
+@implements(proc=NormVecDelay, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class NormVecDelayModel(PyLoihiProcessModel):
+    a_in1 = LavaPyType(PyInPort.VEC_DENSE, np.int32, precision=24)
+    a_in2 = LavaPyType(PyInPort.VEC_DENSE, np.int32, precision=24)
+    s_out = LavaPyType(PyOutPort.VEC_DENSE, np.int32, precision=24)
+    s2_out = LavaPyType(PyOutPort.VEC_DENSE, np.int32, precision=24)
+
+    vth: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    exp: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    v: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    v2: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+    def run_spk(self) -> None:
+        a_in_data1 = self.a_in1.recv()
+        a_in_data2 = self.a_in2.recv()
+
+        vsq = a_in_data1 ** 2
+        self.s2_out.send(vsq)
+
+        self.v2 = self.v
+        self.v = a_in_data1
+
+        output = self.v2 * a_in_data2
+
+        is_spike = np.abs(output) > self.vth
+        sp_out = output * is_spike
+
+        self.s_out.send(sp_out)
+
+
+@implements(proc=InvSqrt, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('float')
+class InvSqrtModelFloat(PyLoihiProcessModel):
+    a_in = LavaPyType(PyInPort.VEC_DENSE, float)
+    s_out = LavaPyType(PyOutPort.VEC_DENSE, float)
+
+    fp_base: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+    def run_spk(self) -> None:
+        a_in_data = self.a_in.recv()
+        sp_out = 1 / (a_in_data ** 0.5)
+
+        self.s_out.send(sp_out)
+
+
+def make_fpinv_table(fp_base):
+    n_bits = 24
+    B = 2**fp_base
+
+    Y_est = np.zeros((n_bits), dtype='int')
+    n_adj = 1.238982962
+
+    for m in range(n_bits):  # span the 24 bits, negate the decimal base
+        Y_est[n_bits - m - 1] = 2 * int(B / (2**((m - fp_base) / 2) * n_adj))
+
+    return Y_est
+
+
+def clz(val):
+    return (24 - (int(np.log2(val)) + 1))
+
+
+def inv_sqrt(s_fp, n_iters=5, b_fraction=12):
+    Y_est = make_fpinv_table(b_fraction)
+    m = clz(s_fp)
+    b_i = int(s_fp)
+    Y_i = Y_est[m]
+    y_i = Y_i // 2
+
+    for i in range(n_iters):
+        b_i = np.right_shift(np.right_shift(b_i * Y_i,
+                                            b_fraction + 1) * Y_i,
+                             b_fraction + 1)
+        Y_i = np.left_shift(3, b_fraction) - b_i
+        y_i = np.right_shift(y_i * Y_i, b_fraction + 1)
+
+    return y_i
+
+
+@implements(proc=InvSqrt, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class InvSqrtModelFP(PyLoihiProcessModel):
```

Comment:
FP could be either Fixed-Pt or Floating-Pt, so I'd favor <Name>Fixed or <Name>FxPt or anything that distinguishes between those. The @tag won't be visible in intellisense.